### ISSUE TITLE
F722 False Positive (Syntax error in forward annotation)

### ISSUE THREAD

**Issue by USSX-Hares** [astral-sh/ruff#26646]
### Summary

```python
type BitwiseOperation = Literal['and', 'or', 'xor']
```

```
F722 Syntax error in forward annotation: Expected an identifier, but found a keyword `or` that cannot be used here
  --> src/chessmaster/util/mongo_query_typing.py:93:40
   |
91 | class ArrayUpdateQuery[T](_ArrayUpdateQuery): pass
92 |
93 | type BitwiseOperation = Literal['and', 'or', 'xor']
   |                                        ^^^^
94 | BitwiseUpdateQuery = TypedDict \
95 | (
```

### Version

ruff 0.15.20

**Comment by ntBre** [comment]
Are you importing this `Literal` from somewhere other than `typing`? If I used `typing.Literal`, I can't reproduce this diagnostic in the playground:

```py
from typing import Literal

type BitwiseOperation = Literal['and', 'or', 'xor']
```

[Playground](https://play.ruff.rs/9b848038-1faa-4e9e-b687-351eaf87d8d5)
 
If it comes from somewhere else, you may need to configure a custom list of [`typing-modules`](https://docs.astral.sh/ruff/settings/#lint_typing-modules). In that case, I think this is related to https://github.com/astral-sh/ruff/issues/11378.

**Comment by USSX-Hares** [comment]
It's `from typing import *`.

**Comment by ntBre** [comment]
Ah, unfortunately Ruff also can't resolve the `Literal` import in that context, so this does seem like the same underlying issue.
